From a13e75564ff3e8102ca3f36385fa0a663aa9b781 Mon Sep 17 00:00:00 2001
From: Bogdan Togorean <bogdan.togorean@analog.com>
Date: Thu, 15 Apr 2021 11:26:07 +0300
Subject: [PATCH 01/15] Enable I2C in PMIC

Signed-off-by: TalPilo <tal.pilo@solid-run.com>
Signed-off-by: Bogdan Togorean <bogdan.togorean@analog.com>
---
 drivers/regulator/pca9450-regulator.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/regulator/pca9450-regulator.c b/drivers/regulator/pca9450-regulator.c
index 247de3e68d81..d0c74f896849 100644
--- a/drivers/regulator/pca9450-regulator.c
+++ b/drivers/regulator/pca9450-regulator.c
@@ -766,6 +766,13 @@ static int pca9450_i2c_probe(struct i2c_client *i2c,
 		return -EINVAL;
 	}
 
+	ret = regmap_write(pca9450->regmap, PCA9450_REG_CONFIG2, 11);
+	if (ret < 0) {
+                dev_err(pca9450->dev, "%s(): write PCA9450_REG_CONFIG2 failed!\n",
+                        __func__);
+                return ret;
+        }
+
 	for (i = 0; i < pca9450->rcnt; i++) {
 		const struct regulator_desc *desc;
 		struct regulator_dev *rdev;
-- 
2.32.0

