From 1ffb9d168af9e135df78ea6e81e6ba0656c52547 Mon Sep 17 00:00:00 2001
From: Bogdan Togorean <bogdan.togorean@analog.com>
Date: Thu, 5 May 2022 11:48:14 +0300
Subject: [PATCH 30/36] Rename control to ADSD

Signed-off-by: Bogdan Togorean <bogdan.togorean@analog.com>
---
 include/uapi/linux/v4l2-controls.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/uapi/linux/v4l2-controls.h b/include/uapi/linux/v4l2-controls.h
index 21e3b433856a..9119621a92fb 100644
--- a/include/uapi/linux/v4l2-controls.h
+++ b/include/uapi/linux/v4l2-controls.h
@@ -202,7 +202,7 @@ enum v4l2_colorfx {
 * The base for the addicmos driver controls.
 * We reserve 16 controls for this driver.
 */
-#define V4L2_CID_USER_ADDICMOS_BASE		(V4L2_CID_USER_BASE + 0x10e0)
+#define V4L2_CID_USER_ADSD_BASE		(V4L2_CID_USER_BASE + 0x10e0)
 
 /* MPEG-class control IDs */
 /* The MPEG controls are applicable to all codec controls
-- 
2.17.1

