From 92e3400cb387b1136d52c05839d2278fffbefb18 Mon Sep 17 00:00:00 2001
From: Bogdan Togorean <bogdan.togorean@analog.com>
Date: Wed, 13 Jul 2022 10:54:29 +0300
Subject: [PATCH 36/40] media: i2c: adsd3500: Implement SET/GET framerate
 control

Signed-off-by: Bogdan Togorean <bogdan.togorean@analog.com>
---
 drivers/media/i2c/adsd3500.c | 30 +++++++++++++++++++++++++++---
 1 file changed, 27 insertions(+), 3 deletions(-)

diff --git a/drivers/media/i2c/adsd3500.c b/drivers/media/i2c/adsd3500.c
index 358a68fa47d6..c498ec7857d1 100644
--- a/drivers/media/i2c/adsd3500.c
+++ b/drivers/media/i2c/adsd3500.c
@@ -206,6 +206,7 @@ static bool adsd3500_regmap_accessible_reg(struct device *dev, unsigned int reg)
 		case GET_IMAGER_CONFIDENCE_TRSHLD:
 		case GET_IMAGER_JBLF_STATE:
 		case GET_IMAGER_JBLF_FILT_SIZE:
+		case GET_FRAMERATE_CMD:
 			return 1;
 		default:
 			return 0;
@@ -742,16 +743,39 @@ static int adsd3500_s_stream(struct v4l2_subdev *subdev, int enable)
 static int adsd3500_g_frame_interval(struct v4l2_subdev *subdev,
 				     struct v4l2_subdev_frame_interval *fi)
 {
+	struct adsd3500 *adsd3500 = to_adsd3500(subdev);
+	uint32_t val;
+	int ret;
+
 	fi->interval.numerator = 1;
-	fi->interval.denominator = 30;
+	fi->interval.denominator = 10;
 
-	return 0;
+	ret = regmap_read(adsd3500->regmap, GET_FRAMERATE_CMD, &val);
+	if (ret < 0)
+		dev_err(adsd3500->dev, "Get FRAMERATE COMMAND failed.\n");
+	else
+		fi->interval.denominator = val;
+
+	return ret;
 }
 
 static int adsd3500_s_frame_interval(struct v4l2_subdev *subdev,
 				     struct v4l2_subdev_frame_interval *fi)
 {
-	return 0;
+	struct adsd3500 *adsd3500 = to_adsd3500(subdev);
+	uint32_t val;
+	int ret;
+
+	val = DIV_ROUND_UP(fi->interval.denominator,  fi->interval.numerator);
+
+	ret = regmap_write(adsd3500->regmap, SET_FRAMERATE_CMD, val);
+	if (ret < 0)
+		dev_err(adsd3500->dev, "Set FRAMERATE COMMAND failed.\n");
+
+	dev_dbg(adsd3500->dev, "Set frame interval to %u / %u\n",
+		fi->interval.numerator, fi->interval.denominator);
+
+	return ret;
 }
 
 static int adsd3500_link_setup(struct media_entity *entity,
-- 
2.37.1

