From e1e02886a4db741039bb1a5e36ec70faa43b8804 Mon Sep 17 00:00:00 2001
From: Bogdan Togorean <bogdan.togorean@analog.com>
Date: Thu, 29 Sep 2022 13:32:58 +0300
Subject: [PATCH 43/50] drivers: media: spi: adsd3100: Add 1 pase mode

Signed-off-by: Bogdan Togorean <bogdan.togorean@analog.com>
---
 drivers/media/spi/adsd3100.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/media/spi/adsd3100.c b/drivers/media/spi/adsd3100.c
index da5b9fe9b961..d2177194dc47 100644
--- a/drivers/media/spi/adsd3100.c
+++ b/drivers/media/spi/adsd3100.c
@@ -178,6 +178,12 @@ static const struct adsd3100_mode_info adsd3100_mode_info_data[] = {
 		.pixel_rate = 488000000,
 		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
 	},
+	{ /* 1 Phase mode for BD */
+		.width = 4096,
+		.height = 192,
+		.pixel_rate = 488000000,
+		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
+	},
 	{
 		.width = 4096,
 		.height = 256,
-- 
2.39.0

