From 5452e2b99136868473c202b0ee59cc1f46e859ea Mon Sep 17 00:00:00 2001
From: Bogdan Togorean <bogdan.togorean@analog.com>
Date: Fri, 25 Nov 2022 16:32:48 +0200
Subject: [PATCH 47/47] drivers: media: adsd3500: Add support fo 3F3P 3F2P VGA
 & QVGA RAW

Signed-off-by: Bogdan Togorean <bogdan.togorean@analog.com>
---
 drivers/media/i2c/adsd3500.c | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/drivers/media/i2c/adsd3500.c b/drivers/media/i2c/adsd3500.c
index 2dee4602a506..3f118fd9bffb 100644
--- a/drivers/media/i2c/adsd3500.c
+++ b/drivers/media/i2c/adsd3500.c
@@ -211,6 +211,20 @@ static const struct adsd3500_mode_info adsd3500_mode_info_data[] = {
 		.code = MEDIA_BUS_FMT_Y12_1X12,
 		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
 	},
+	{ //RAW12 12BPP * 6 subframes ADSD3030 256x320x6
+		.width = 1024,
+		.height = 480,
+		.pixel_rate = 488000000,
+		.code = MEDIA_BUS_FMT_Y12_1X12,
+		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
+	},
+	{ //RAW12 12BPP * 9 subframes ADSD3030 256x320x9
+		.width = 1024,
+		.height = 720,
+		.pixel_rate = 488000000,
+		.code = MEDIA_BUS_FMT_Y12_1X12,
+		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
+	},
 	{ //RAW12 12BPP * 3 subframes ADSD3030 512x640x3
 		.width = 1024,
 		.height = 960,
@@ -218,6 +232,13 @@ static const struct adsd3500_mode_info adsd3500_mode_info_data[] = {
 		.code = MEDIA_BUS_FMT_Y12_1X12,
 		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
 	},
+	{ //RAW12 12BPP * 6 subframes ADSD3030 512x640x6
+		.width = 1024,
+		.height = 1920,
+		.pixel_rate = 488000000,
+		.code = MEDIA_BUS_FMT_Y12_1X12,
+		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
+	},
 	{ //RAW12 12BPP * 9 subframes ADSD3030 512x640x9
 		.width = 1024,
 		.height = 2880,
-- 
2.38.1

