// @adi-ctx/harmonic-core-components/dist/esm/hmc-tab-panel.entry.js
import { r as registerInstance, h, H as Host, g as getElement } from './index-9bcb3512.js';

const tabPanelCss = ":host{visibility:hidden;display:none;contain:layout;overflow-y:hidden;width:100%}:host(.selected){visibility:visible;display:flex}:host(:focus-visible){outline-offset:2px;outline:2px solid var(--hmc-color-interactive-focus, var(--hmc-color-system-highlight-default, var(--hmc-color-purple-500, #b16ee0)))}";

const PANEL_ID = 'hmc-panel-id-';
const HmcTabPanel = class {
  constructor(hostRef) {
    registerInstance(this, hostRef);
    this.name = undefined;
    this.isSelected = false;
  }
  handleKeyEvent(ev) {
    ev.stopPropagation();
  }
  componentWillLoad() {
    this._id = this.el.id.length > 0 ? this.el.id : PANEL_ID + panelCount++;
  }
  render() {
    const { isSelected } = this;
    const classList = {
      'title-container': true,
      'selected': isSelected,
    };
    return (h(Host, { role: "tabpanel", tabindex: "0", class: classList, id: this._id }, h("slot", null)));
  }
  get el() { return getElement(this); }
};
let panelCount = 0;
HmcTabPanel.style = tabPanelCss;

export { HmcTabPanel as hmc_tab_panel };
