// @adi-ctx/harmonic-core-components/dist/esm/hmc-banner-message.entry.js
import { r as registerInstance, c as createEvent, h, H as Host } from './index-9bcb3512.js';

const bannerMessageCss = ":host{display:block;width:100%;color:var(--hmc-color-notification-info-default-text, var(--hmc-color-system-info-default, var(--hmc-color-content-default, var(--hmc-color-gray-black, #101820))));font-family:var(--font-family-body, \"Inter\", sans-serif);font-size:var(--font-size-medium, 16px);line-height:24px}.banner{display:flex;justify-content:space-between;width:100%;box-sizing:border-box;padding:12px 16px;background-color:var(--hmc-color-notification-info-default-background, var(--hmc-color-system-info-silent, var(--hmc-color-gray-100, #f0f1f3)))}.banner .main-content{padding-top:3px}.banner .content-container hmc-icon{color:var(--hmc-color-notification-info-default-graphic, var(--hmc-color-system-info-default, var(--hmc-color-content-default, var(--hmc-color-gray-black, #101820))))}.banner.high-density{padding:8px 16px;font-size:14px;line-height:20px}.banner.high-density .content-container{padding-top:1px}.banner.high-density .main-content{padding-top:0}.banner.success{color:var(--hmc-color-notification-success-default-text, var(--hmc-color-content-default, var(--hmc-color-gray-black, #101820)));background-color:var(--hmc-color-notification-success-default-background, var(--hmc-color-system-success-silent, var(--hmc-color-green-100, #e4f6ef)))}.banner.success .content-container hmc-icon{color:var(--hmc-color-notification-success-default-graphic, var(--hmc-color-system-success-default, var(--hmc-color-green-500, #2e9e6f)))}.banner.warning{color:var(--hmc-color-notification-warning-default-text, var(--hmc-color-content-default, var(--hmc-color-gray-black, #101820)));background-color:var(--hmc-color-notification-warning-default-background, var(--hmc-color-system-warning-silent, var(--hmc-color-orange-100, #feefe7)))}.banner.warning .content-container hmc-icon{color:var(--hmc-color-notification-warning-default-graphic, var(--hmc-color-system-warning-default, var(--hmc-color-orange-500, #e76423)))}.banner.error{color:var(--hmc-color-notification-danger-default-text, var(--hmc-color-system-danger-default, var(--hmc-color-red-700, #c81a28)));background-color:var(--hmc-color-notification-danger-default-background, var(--hmc-color-system-danger-silent, var(--hmc-color-red-100, #feeeef)))}.banner.error .content-container hmc-icon{color:var(--hmc-color-notification-danger-default-graphic, var(--hmc-color-system-danger-default, var(--hmc-color-red-700, #c81a28)))}.content-container,.action-container{display:flex;gap:8px}.content-container{margin-right:12px}.content-container hmc-icon{padding-top:2px;height:100%}";

const statusIconMap = {
  low: {
    success: 'check-round-m',
    warning: 'warning-m',
    error: 'error-m',
    info: 'info-m',
  },
  high: {
    success: 'check-circle-s',
    warning: 'warning-s',
    error: 'error-s',
    info: 'info-s',
  },
};
const HmcBannerMessage = class {
  constructor(hostRef) {
    registerInstance(this, hostRef);
    this.actionClicked = createEvent(this, "actionClicked", 7);
    this.bannerCloseRequested = createEvent(this, "bannerCloseRequested", 7);
    this.status = 'info';
    this.density = 'low';
    this.actionText = '';
    this.isClosable = false;
    this.withIcon = true;
    this.iconName = undefined;
  }
  render() {
    const { actionText, density, iconName, isClosable, status, withIcon, } = this;
    const classList = {
      'banner': true,
      [status]: true,
      'high-density': density === 'high',
    };
    const icon = (iconName || statusIconMap[density][status]);
    return (h(Host, null, h("div", { class: classList }, h("div", { class: "content-container" }, withIcon && h("hmc-icon", { name: icon }), h("div", { class: "main-content" }, h("slot", null))), h("div", { class: "action-container" }, actionText !== '' && (h("hmc-button", { size: density === 'low' ? 'sm' : 'xs', kind: "secondary", onClick: () => this.actionClicked.emit() }, actionText)), isClosable && (h("hmc-button", { "icon-only": true, size: density === 'low' ? 'sm' : 'xs', kind: "ghost", onClick: () => this.bannerCloseRequested.emit() }, h("hmc-icon", { name: "close-m" })))))));
  }
};
HmcBannerMessage.style = bannerMessageCss;

export { HmcBannerMessage as hmc_banner_message };
