// @adi-ctx/harmonic-core-components/dist/collection/components/container-header/container-header.stories.js
import { html } from 'lit-html';

import documentation from './container-header-documentation.mdx';
import readme from './readme.md';

export default {
  title: 'Components/Container Header',
  parameters: { notes: readme, docs: { page: documentation } },
  argTypes: {
    density: {
      control: {
        type: 'inline-radio',
        options: ['low', 'medium', 'high'],
      },
    },
  },
};

const style = html`
  <style>
    .main {
      padding: 16px;
      background-color: var(--hmc-color-layout-canvas);
    }
    .container {
      background-color: var(--hmc-color-layout-container);
      padding: 16px 24px;
      width: 250px;
      border-radius: 2px;
    }
  </style>
`;

const Template = ({ density, hasIcon, subtitle, heading }) => {
  return html`
    ${style}
    <div class="main">
      <div class="container">
        <hmc-container-header
          density=${density}
          heading=${heading}
          subtitle=${subtitle}
        >
          ${hasIcon &&
          html`<hmc-icon slot="icon" name="placeholder-m"></hmc-icon>`}
        </hmc-container-header>
      </div>
    </div>
  `;
};

export const ContainerHeader = Template.bind({});
ContainerHeader.args = {
  heading: 'Container Title',
  subtitle: 'Container Subtitle',
  hasIcon: true,
  density: 'medium',
};
