// @adi-ctx/harmonic-core-components/dist/collection/components/checkbox/checkbox.stories.js
import { html } from 'lit-html';

import documentation from './checkbox-documentation.mdx';
import readme from './readme.md';

export default {
  title: 'Components/Checkbox',
  parameters: {
    notes: readme,
    docs: {
      page: documentation,
    },
  },
  args: {
    label: 'Label',
    isChecked: false,
    isDisabled: false,
    isReadonly: false,
    size: 'md',
  },
  argTypes: {
    size: {
      control: {
        type: 'inline-radio',
        options: ['xs', 'md'],
      },
    },
    isReadonly: {
      table: {
        disable: true,
      },
    },
    onChanged: { action: 'changed' },
  },
};
const style = html`<style>
  hmc-checkbox {
    margin-right: 72px;
    width: 10%;
  }
  .grid {
    width: 100%;
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(150px, 1fr));
  }
</style>`;
export const Playground = ({
  isChecked,
  isDisabled,
  isReadonly,
  label,
  size,
  onChanged,
}) =>
  html`
    <hmc-checkbox
      label=${label}
      checked=${isChecked}
      disabled=${isDisabled}
      readonly=${isReadonly}
      size=${size}
      @changed=${onChanged}
    ></hmc-checkbox>
  `;
Playground.argTypes = {
  isReadonly: {
    table: {
      disable: false,
    },
  },
};
export const States = ({ size }) =>
  html`
    ${style}
    <div class="grid">
      <hmc-checkbox
        size=${size}
        label="Unchecked"
        readonly="true"
      ></hmc-checkbox>

      <hmc-checkbox
        size=${size}
        label="Checked"
        readonly="true"
        checked="true"
      ></hmc-checkbox>
      <hmc-checkbox
        size=${size}
        label="Disabled"
        disabled="true"
      ></hmc-checkbox>
      <hmc-checkbox
        size=${size}
        label="Disabled"
        disabled="true"
        checked="true"
      ></hmc-checkbox>
    </div>
  `;

export const Sizes = ({ onChanged }) =>
  html`
    ${style}
    <div>
      <hmc-checkbox
        label="Medium"
        size="md"
        @changed=${onChanged}
      ></hmc-checkbox>
      <hmc-checkbox
        label="Extra Small"
        size="xs"
        @changed=${onChanged}
      ></hmc-checkbox>
    </div>
  `;

Sizes.argTypes = {
  size: {
    table: {
      disable: true,
    },
  },
  label: {
    table: {
      disable: true,
    },
  },
  isDisabled: {
    table: {
      disable: true,
    },
  },
  isChecked: {
    table: {
      disable: true,
    },
  },
  onChanged: {
    table: {
      disable: true,
    },
  },
};
States.argTypes = {
  ...Sizes.argTypes,
  size: {
    table: {
      disable: false,
    },
  },
};
