// @adi-ctx/harmonic-core-components/dist/collection/components/theme-provider/theme-provider.stories.js
import { html } from 'lit-html';
import { styleMap } from 'lit-html/directives/style-map.js';

import readme from './readme.md';
import document from './theme-provider-documentation.mdx';

export default {
  title: 'Components/Theme Provider',
  parameters: {
    notes: readme,
    docs: {
      page: document,
    },
  },
};

const style = html`<style>
  .container {
    width: 100vw;
    height: 100vh;
    background-color: var(--hmc-color-layout-canvas);
    padding: 40px;
  }
  hmc-card {
    max-width: 500px;
  }
  .card-content {
    display: flex;
    flex-direction: column;
    gap: 16px;
  }
</style>`;
export const Playground = ({ theme }) => {
  return html`
    ${style}
    <hmc-theme-provider theme=${theme}
      ><div class="container">
        <hmc-card>
          <hmc-container-header
            slot="card-header"
            heading="Heading"
            subtitle="Subtitle"
          ></hmc-container-header>
          <div class="card-content">
            <hmc-banner-message status="info">
              Banner message content
            </hmc-banner-message>
            <hmc-banner-message status="success">
              Banner message content
            </hmc-banner-message>
            <hmc-banner-message status="warning">
              Banner message content
            </hmc-banner-message>
            <hmc-banner-message status="error">
              Banner message content
            </hmc-banner-message>
            <hmc-segmented-control value="1">
              <hmc-segmented-button value="1" full-width="true"
                >Segment One </hmc-segmented-button
              ><hmc-segmented-button value="2" full-width="true"
                >Segment Two </hmc-segmented-button
              ><hmc-segmented-button value="3" full-width="true"
                >Segment Three
              </hmc-segmented-button>
            </hmc-segmented-control>

            <hmc-input-field
              placeholder="Placeholder"
              status="info"
              label="Input field"
              helper-text="Info text"
            >
            </hmc-input-field>
            <hmc-input-field
              value="filled"
              status="error"
              label="Input field"
              helper-text="Info text"
            >
            </hmc-input-field>
          </div>
          <div
            slot="card-footer"
            class="card-footer-wrapper"
            style=${styleMap({
              'display': 'flex',
              'justify-content': 'flex-end',
              'align-items': 'center',
              'gap': '12px',
            })}
          >
            <hmc-button kind="tertiary" size="sm">Button Label</hmc-button>
            <hmc-button kind="secondary" size="sm">Button Label</hmc-button>
          </div>
        </hmc-card>
      </div></hmc-theme-provider
    >
  `;
};
Playground.args = {
  theme: 'dark',
};
Playground.argTypes = {
  theme: {
    control: {
      type: 'inline-radio',
      options: ['light', 'dark'],
    },
  },
};
