// @adi-ctx/harmonic-core-components/dist/collection/components/toast-message/toast-message.js
import { Host, h, } from '@stencil/core';
const statusIconMap = {
  low: {
    success: 'check-round-m',
    warning: 'warning-m',
    error: 'error-m',
    info: 'info-m',
  },
  high: {
    success: 'check-circle-s',
    warning: 'warning-s',
    error: 'error-s',
    info: 'info-s',
  },
};
/**
 * @slot - An unnamed slot is used to pass the main text content of the toast.
 */
export class HmcToastMessage {
  constructor() {
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    this.timeoutRef = setTimeout(() => { }, null);
    this.status = 'info';
    this.density = 'low';
    this.actionText = '';
    this.hasCloseButton = false;
    this.withIcon = true;
    this.iconName = undefined;
    this.isOpen = false;
    this.timeout = 5000;
  }
  handleOpen(newValue, oldValue) {
    if (!oldValue && newValue) {
      this.initiateClosing();
    }
  }
  initiateClosing() {
    if (!this.toastClosing.emit().defaultPrevented) {
      this.startToastCloseTimer();
    }
  }
  closeToast() {
    this.toastClosed.emit();
    this.isOpen = false;
  }
  startToastCloseTimer() {
    if (this.isOpen) {
      this.timeoutRef = setTimeout(() => {
        this.closeToast();
      }, this.timeout);
    }
  }
  stopToastCloseTimer() {
    clearTimeout(this.timeoutRef);
  }
  componentDidLoad() {
    if (this.isOpen) {
      this.initiateClosing();
    }
    this.el.addEventListener('mouseover', () => this.stopToastCloseTimer());
    this.el.addEventListener('mouseout', () => this.startToastCloseTimer());
  }
  render() {
    const { actionText, density, iconName, hasCloseButton, status, withIcon, isOpen, } = this;
    const classList = {
      'toast': true,
      [status]: true,
      'high-density': density === 'high',
      'open': isOpen,
    };
    const icon = (iconName || statusIconMap[density][status]);
    return (h(Host, { class: isOpen ? 'open' : '' }, h("div", { class: classList, role: "status" }, h("div", { class: "content-container" }, withIcon && h("hmc-icon", { name: icon }), h("div", { class: "main-content" }, h("slot", null))), h("div", { class: "action-container" }, actionText !== '' && (h("hmc-button", { size: density === 'low' ? 'sm' : 'xs', kind: "secondary", onClick: () => this.actionClicked.emit() }, actionText)), hasCloseButton && (h("hmc-button", { "icon-only": true, size: density === 'low' ? 'sm' : 'xs', kind: "ghost", onClick: () => this.closeToast() }, h("hmc-icon", { name: "close-m" })))))));
  }
  static get is() { return "hmc-toast-message"; }
  static get encapsulation() { return "shadow"; }
  static get originalStyleUrls() {
    return {
      "$": ["toast-message.scss"]
    };
  }
  static get styleUrls() {
    return {
      "$": ["toast-message.css"]
    };
  }
  static get properties() {
    return {
      "status": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "MessageStatusType",
          "resolved": "\"error\" | \"info\" | \"success\" | \"warning\"",
          "references": {
            "MessageStatusType": {
              "location": "import",
              "path": "../type/status.type"
            }
          }
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": "Status to convery to the user with the message."
        },
        "attribute": "status",
        "reflect": false,
        "defaultValue": "'info'"
      },
      "density": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "'low' | 'high'",
          "resolved": "\"high\" | \"low\"",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": "Density of the information conveyed. Higher density will occupy less space in the\ninterface for the same information."
        },
        "attribute": "density",
        "reflect": false,
        "defaultValue": "'low'"
      },
      "actionText": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": "Text label for the optional button. Providing a value will cause a button to render."
        },
        "attribute": "action-text",
        "reflect": false,
        "defaultValue": "''"
      },
      "hasCloseButton": {
        "type": "boolean",
        "mutable": false,
        "complexType": {
          "original": "boolean",
          "resolved": "boolean",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": "If set `true`, the toast will display a button with a close icon. When the user clicks\nthis button a `toastClosed` event will fire and the toast will close immediately."
        },
        "attribute": "close-button",
        "reflect": false,
        "defaultValue": "false"
      },
      "withIcon": {
        "type": "boolean",
        "mutable": false,
        "complexType": {
          "original": "boolean",
          "resolved": "boolean",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": "Toggles the display of an icon to the left of the toast main text."
        },
        "attribute": "with-icon",
        "reflect": false,
        "defaultValue": "true"
      },
      "iconName": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": "Allows the optional override of the icon to the left of the text. If not provided, the icon will be\ndetermined based on status. Small icons should be used with the high density while medium icons work best with low."
        },
        "attribute": "icon-name",
        "reflect": false
      },
      "isOpen": {
        "type": "boolean",
        "mutable": true,
        "complexType": {
          "original": "boolean",
          "resolved": "boolean",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "open",
        "reflect": true,
        "defaultValue": "false"
      },
      "timeout": {
        "type": "number",
        "mutable": false,
        "complexType": {
          "original": "number",
          "resolved": "number",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "timeout",
        "reflect": false,
        "defaultValue": "5000"
      }
    };
  }
  static get events() {
    return [{
        "method": "actionClicked",
        "name": "actionClicked",
        "bubbles": true,
        "cancelable": true,
        "composed": true,
        "docs": {
          "tags": [],
          "text": "This event will fire when the user clicks the optional button which appears when `actionText` is provided."
        },
        "complexType": {
          "original": "any",
          "resolved": "any",
          "references": {}
        }
      }, {
        "method": "toastClosing",
        "name": "toastClosing",
        "bubbles": true,
        "cancelable": true,
        "composed": true,
        "docs": {
          "tags": [],
          "text": "This event fires when the toast opens and the timeout counter begins. The app can use preventDefault() on the event\nto cancel the timeout."
        },
        "complexType": {
          "original": "any",
          "resolved": "any",
          "references": {}
        }
      }, {
        "method": "toastClosed",
        "name": "toastClosed",
        "bubbles": true,
        "cancelable": true,
        "composed": true,
        "docs": {
          "tags": [],
          "text": "This event fires when the user clicks the close button or the closing timeout completes. The app can then respond to this and remove the\ntoast from the UI."
        },
        "complexType": {
          "original": "any",
          "resolved": "any",
          "references": {}
        }
      }];
  }
  static get elementRef() { return "el"; }
  static get watchers() {
    return [{
        "propName": "isOpen",
        "methodName": "handleOpen"
      }];
  }
}
