// @adi-ctx/harmonic-core-components/dist/cjs/hmc-card.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-d9134559.js');

const cardCss = ":host{display:block;width:100%}.card{color:var(--hmc-color-content-default, var(--hmc-color-gray-black, #101820));font-family:var(--font-family-body, \"Inter\", sans-serif);font-size:var(--font-size-medium, 16px);line-height:24px;display:flex;flex-direction:column;width:100%;height:var(--hmc-card-height, auto);box-sizing:border-box;background-color:var(--hmc-color-layout-container, var(--hmc-color-gray-white, #fff));border-radius:var(--hmc-card-border-radius, 8px);border:var(--hmc-card-border-width, 1px) solid var(--hmc-color-layout-divider-default, var(--hmc-color-gray-300, #b7bbc3))}.card.high-density .card-content{padding:var(--hmc-card-content-padding, 16px);font-size:14px;line-height:20px}.card.high-density .p-x{padding:0 16px}.card.high-density .card-footer-container{padding:16px 0}.card.high-density .card-header-container{padding-top:6px}.card-content{flex:1;padding:var(--hmc-card-content-padding, 20px)}.card-header-right{display:flex;align-items:center;gap:8px;justify-content:flex-end;flex-direction:row}.card-header-container{box-sizing:border-box;display:flex;align-items:flex-start;justify-content:space-between;padding-top:12px;border-bottom:1px solid var(--hmc-color-layout-divider-default, var(--hmc-color-gray-300, #b7bbc3))}.card-footer-container{padding:20px 0px;border-top:solid 1px var(--hmc-color-layout-divider-default, var(--hmc-color-gray-300, #b7bbc3))}.p-x{padding:0 20px}";

const HmcCard = class {
  constructor(hostRef) {
    index.registerInstance(this, hostRef);
    this.density = 'medium';
    this.hasHeader = true;
    this.hasFooter = false;
  }
  componentWillLoad() {
    this.hasHeader = Boolean(this.el.querySelector('[slot="card-header"]'));
    this.hasFooter = Boolean(this.el.querySelector('[slot="card-footer"]'));
  }
  renderHeader() {
    const rootClass = {
      'card-header-container': true,
      'high-density': this.density === 'high',
    };
    return (index.h("div", { class: "p-x" }, index.h("div", { class: rootClass }, index.h("slot", { name: "card-header" }), index.h("div", { class: "card-header-right" }, index.h("slot", { name: "card-header-actions" })))));
  }
  renderFooter() {
    return (index.h("div", { class: "p-x" }, index.h("div", { class: "card-footer-container" }, index.h("slot", { name: "card-footer" }))));
  }
  render() {
    const { density, hasHeader, hasFooter } = this;
    const rootClass = {
      'card': true,
      'high-density': density === 'high',
    };
    return (index.h(index.Host, null, index.h("div", { class: rootClass }, hasHeader && this.renderHeader(), index.h("div", { class: "card-content" }, index.h("slot", null)), hasFooter && this.renderFooter())));
  }
  get el() { return index.getElement(this); }
};
HmcCard.style = cardCss;

exports.hmc_card = HmcCard;
