// @adi-ctx/harmonic-core-components/dist/cjs/hmc-modal-overlay.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-d9134559.js');

const modalOverlayCss = ".overlay{display:flex;justify-content:center;align-items:center;height:100%;width:100%;position:fixed;left:0;top:0;z-index:var(--z-index-overlay, 8000);background-color:var(--hmc-color-layout-overlay, rgba(94, 103, 115, 0.75))}";

const HmcModalOverlay = class {
  constructor(hostRef) {
    index.registerInstance(this, hostRef);
    this.overlayClick = index.createEvent(this, "overlayClick", 7);
    this.isOpen = false;
  }
  onOverlayClickHandler({ target }) {
    if (target === this.overlayEl) {
      this.overlayClick.emit();
    }
  }
  render() {
    const { isOpen } = this;
    const rootClass = {
      'overlay': true,
    };
    return isOpen && (index.h("div", { ref: ref => (this.overlayEl = ref), class: rootClass, onClick: (e) => this.onOverlayClickHandler(e) }, index.h("slot", null)));
  }
};
HmcModalOverlay.style = modalOverlayCss;

exports.hmc_modal_overlay = HmcModalOverlay;
