// @adi-ctx/harmonic-core-components/dist/collection/components/progress-bar/progress-bar.stories.js
import { html } from 'lit-html';

import document from './progress-bar-documentation.mdx';
import readme from './readme.md';

export default {
  title: 'Components/Progress Bar',
  parameters: {
    notes: readme,
    docs: {
      page: document,
    },
  },
  argTypes: {
    size: {
      control: {
        type: 'inline-radio',
        options: ['sm', 'md'],
      },
    },
    status: {
      control: {
        type: 'inline-radio',
        options: ['info', 'success', 'error'],
      },
    },
    value: {
      control: {
        type: 'number',
      },
    },
  },
};

export const Playground = ({
  indeterminate,
  inlineMessage,
  label,
  maxValue,
  showPercentage,
  size,
  slottedButton,
  status,
  value,
}) => {
  return html`
    <style>
      .container {
        width: 300px;
      }
    </style>
    <div class="container">
      <hmc-progress-bar
        value="${value}"
        max-value="${maxValue}"
        status="${status}"
        ?indeterminate=${indeterminate}
        ?show-percentage=${showPercentage}
        size="${size}"
        label="${label}"
        inline-message="${inlineMessage}"
      >
        ${slottedButton
          ? html` <hmc-button
              size="xs"
              kind="ghost"
              icon-only
              slot="action-button"
            >
              <hmc-icon name="placeholder-s"></hmc-icon>
            </hmc-button>`
          : ''}
      </hmc-progress-bar>
    </div>
  `;
};
Playground.args = {
  size: 'md',
  status: 'info',
  label: 'Label',
  value: 0,
  maxValue: 100,
  indeterminate: false,
  showPercentage: true,
  inlineMessage: 'Message',
  slottedButton: true,
};
