// @adi-ctx/harmonic-core-components/dist/collection/components/dropdown-group/dropdown-group.stories.js
import { html } from 'lit-html';

import documentation from './dropdown-group-documentation.mdx';
import readme from './readme.md';

export default {
  title: 'Components/Dropdown Menu',
  parameters: {
    notes: readme,
    docs: {
      page: documentation,
    },
  },
  argTypes: {
    size: {
      control: {
        type: 'inline-radio',
        options: ['xs', 'sm', 'md'],
      },
    },
    onSelection: { action: 'hmcDropdownSelection' },
  },
};
const style = html` <style>
  .container {
    width: 250px;
  }
</style>`;
const Template = ({ title, size, onSelection }) => {
  return html`${style}

    <div class="container">
      <hmc-dropdown-group group-title=${title} size=${size}>
        <hmc-dropdown-option
          value="option1"
          selected="false"
          disabled="false"
          size=${size}
          multiple="false"
          @hmcDropdownSelection=${onSelection}
        >
          Idle
        </hmc-dropdown-option>
        <hmc-dropdown-option
          value="option2"
          selected="true"
          disabled="false"
          size=${size}
          multiple="false"
          @hmcDropdownSelection=${onSelection}
        >
          Selected
        </hmc-dropdown-option>
        <hmc-dropdown-option
          value="option3"
          selected="false"
          disabled="false"
          size=${size}
          multiple="true"
          @hmcDropdownSelection=${onSelection}
        >
          Idle (Multi-Choice)
        </hmc-dropdown-option>
      </hmc-dropdown-group>
    </div>`;
};
export const DropdownGroup = Template.bind({});
DropdownGroup.args = {
  size: 'md',
  title: 'Dropdown Group',
};
