// @adi-ctx/harmonic-core-components/dist/collection/components/badge/badge.stories.js
import { html } from 'lit-html';

import readme from './readme.md';
import document from './badge-documentation.mdx';

export default {
  title: 'Components/Badge',
  parameters: {
    notes: readme,
    docs: {
      page: document,
    },
  },
  argTypes: {
    kind: {
      control: {
        type: 'inline-radio',
        options: ['digits', 'dots'],
      },
    },
    status: {
      control: {
        type: 'inline-radio',
        options: ['urgent', 'information'],
      },
    },
    number: {
      control: {
        type: 'number',
      },
    },
    maxNumber: {
      control: {
        type: 'number',
      },
    },
  },
};

const style = html`
  <style>
    .flex {
      display: flex;
      gap: 16px;
      width: max-content;
      align-items: flex-end;
    }
  </style>
`;
export const Playground = ({ kind, status, number, maxNumber }) => {
  return html`
    <hmc-badge
      status="${status}"
      kind="${kind}"
      number="${number}"
      max-number="${maxNumber}"
    ></hmc-badge>
  `;
};
Playground.args = {
  kind: 'digits',
  status: 'urgent',
  number: 1,
  maxNumber: 999,
};

export const Urgent = () => {
  return html`
    ${style}

    <div class="flex">
      <hmc-badge status="urgent" kind="digits" number="1"></hmc-badge>

      <hmc-badge status="urgent" kind="digits" number="11"></hmc-badge>

      <hmc-badge status="urgent" kind="digits" number="111"></hmc-badge>

      <hmc-badge status="urgent" kind="digits" number="1111"></hmc-badge>

      <hmc-badge status="urgent" kind="dots" number="1111"></hmc-badge>
    </div>
  `;
};

Urgent.argTypes = {
  kind: {
    table: {
      disable: true,
    },
  },
  status: {
    table: {
      disable: true,
    },
  },
  number: {
    table: {
      disable: true,
    },
  },
  maxNumber: {
    table: {
      disable: true,
    },
  },
};

export const Information = () => {
  return html`
    ${style}

    <div class="flex">
      <hmc-badge status="information" kind="digits" number="1"></hmc-badge>

      <hmc-badge status="information" kind="digits" number="11"></hmc-badge>

      <hmc-badge status="information" kind="digits" number="111"></hmc-badge>

      <hmc-badge status="information" kind="digits" number="1111"></hmc-badge>

      <hmc-badge status="information" kind="dots" number="1111"></hmc-badge>
    </div>
  `;
};

Information.argTypes = {
  kind: {
    table: {
      disable: true,
    },
  },
  status: {
    table: {
      disable: true,
    },
  },
  number: {
    table: {
      disable: true,
    },
  },
  maxNumber: {
    table: {
      disable: true,
    },
  },
};
