// @adi-ctx/harmonic-core-components/dist/collection/components/spinner/spinner.js
import { Host, h } from '@stencil/core';
/**
 * @slot - An unnamed slot is used to pass in visually hidden text for screen readers.
 */
export class HmcSpinner {
  constructor() {
    this.size = 'sm';
    this.kind = 'primary';
  }
  render() {
    const { kind, size } = this;
    const classes = {
      spinner: true,
      [kind]: true,
      [size]: true,
    };
    return (h(Host, null, h("div", { class: classes, role: "status" }, h("span", { class: "visually-hidden" }, h("slot", null)))));
  }
  static get is() { return "hmc-spinner"; }
  static get encapsulation() { return "shadow"; }
  static get originalStyleUrls() {
    return {
      "$": ["./spinner.scss"]
    };
  }
  static get styleUrls() {
    return {
      "$": ["spinner.css"]
    };
  }
  static get properties() {
    return {
      "size": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "SpinnerSize",
          "resolved": "\"md\" | \"sm\"",
          "references": {
            "SpinnerSize": {
              "location": "import",
              "path": "../type/size.type"
            }
          }
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": "The size of the spinner.\nAvailable options are: 'sm', and 'md'\nDefault: 'sm'."
        },
        "attribute": "size",
        "reflect": false,
        "defaultValue": "'sm'"
      },
      "kind": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "SpinnerVariant",
          "resolved": "\"inverse\" | \"primary\" | \"secondary\" | \"tertiary\"",
          "references": {
            "SpinnerVariant": {
              "location": "import",
              "path": "../type/variant.type"
            }
          }
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": "The kind of spinner.\nAvailable options are: 'primary', 'secondary' and 'tertiary'\nDefault: 'primary'."
        },
        "attribute": "kind",
        "reflect": false,
        "defaultValue": "'primary'"
      }
    };
  }
}
