From c8d8865fa8b9d2b54f6e760f2e068d7119fe46ae Mon Sep 17 00:00:00 2001
From: Bogdan Togorean <bogdan.togorean@analog.com>
Date: Tue, 23 Aug 2022 16:39:27 +0300
Subject: [PATCH 41/50] drivers: media: i2c: adsd3500: Add ADSD30303
 resolutions

Signed-off-by: Bogdan Togorean <bogdan.togorean@analog.com>
---
 drivers/media/i2c/adsd3500.c | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/drivers/media/i2c/adsd3500.c b/drivers/media/i2c/adsd3500.c
index 5668975bc436..e6931320a871 100644
--- a/drivers/media/i2c/adsd3500.c
+++ b/drivers/media/i2c/adsd3500.c
@@ -162,6 +162,34 @@ static const struct adsd3500_mode_info adsd3500_mode_info_data[] = {
 		.code = MEDIA_BUS_FMT_SBGGR12_1X12,
 		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
 	},
+	{ //RAW12 12BPP ADSD3030
+		.width = 512,
+		.height = 640,
+		.pixel_rate = 488000000,
+		.code = MEDIA_BUS_FMT_SBGGR12_1X12,
+		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
+	},
+	{ //RAW8 16BPP ADSD3030
+		.width = 1024,
+		.height = 640,
+		.pixel_rate = 488000000,
+		.code = MEDIA_BUS_FMT_SBGGR8_1X8,
+		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
+	},
+	{ //RAW8 24BPP ADSD3030
+		.width = 1536,
+		.height = 640,
+		.pixel_rate = 488000000,
+		.code = MEDIA_BUS_FMT_SBGGR8_1X8,
+		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
+	},
+	{ //RAW8 32BPP ADSD3030
+		.width = 2048,
+		.height = 640,
+		.pixel_rate = 488000000,
+		.code = MEDIA_BUS_FMT_SBGGR8_1X8,
+		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
+	},
 	{ //RAW12 1 Phase / Frame
 		.width = 1024,
 		.height = 1024,
-- 
2.39.0

