From b3f182add9088f4a6e206f657c44bc062951a3be Mon Sep 17 00:00:00 2001
From: Bogdan Togorean <bogdan.togorean@analog.com>
Date: Fri, 25 Nov 2022 16:30:12 +0200
Subject: [PATCH 46/50] drivers: media: i2c: adsd3500.c: Allow zero BPP for
 depth

Signed-off-by: Bogdan Togorean <bogdan.togorean@analog.com>
---
 drivers/media/i2c/adsd3500.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/media/i2c/adsd3500.c b/drivers/media/i2c/adsd3500.c
index b6a7892bb34c..2dee4602a506 100644
--- a/drivers/media/i2c/adsd3500.c
+++ b/drivers/media/i2c/adsd3500.c
@@ -291,8 +291,8 @@ static int adsd3500_power_on(struct device *dev)
 
 	write_cmd = SET_IMAGER_MODE_CMD | SET_IMAGER_MODE(operating_mode->val);
 
-	write_val |= SET_IMAGER_MODE_DEPTH_EN(depth_en->val);
-	write_val |= SET_IMAGER_MODE_DEPTH_BITS(config.nr_depth_bits ? 6 - config.nr_depth_bits: 0);
+	write_val |= SET_IMAGER_MODE_DEPTH_EN(config.nr_depth_bits ? depth_en->val : 0);
+	write_val |= SET_IMAGER_MODE_DEPTH_BITS(config.nr_depth_bits ? 6 - config.nr_depth_bits : 7);
 	
 	write_val |= SET_IMAGER_MODE_AB_EN(config.nr_ab_bits ? 1 : 0);
 	write_val |= SET_IMAGER_MODE_AB_BITS(config.nr_ab_bits ? 6 - config.nr_ab_bits : 0);
@@ -474,10 +474,10 @@ static const struct v4l2_ctrl_config adsd3500_ctrl_depth_bits = {
 	.id			= V4L2_CID_ADSD3500_DEPTH_BITS,
 	.name		= "Phase / Depth Bits",
 	.type		= V4L2_CTRL_TYPE_INTEGER_MENU,
-	.def		= 2,
-	.min		= 2,
+	.def		= 4,
+	.min		= 0,
 	.max		= 6,
-	.menu_skip_mask = 0x03,
+	.menu_skip_mask = 0x02,
 	.qmenu_int	= nr_bits_qmenu,
 };
 
-- 
2.39.0

